vcd/resource_vcd_org_vdc.go

```diff
@@ -311,38 +311,54 @@ func resourceVcdVdcRead(d *schema.ResourceData, meta interface{}) error {
 func setOrgVdcData(d *schema.ResourceData, vcdClient *VCDClient, adminOrg *govcd.AdminOrg, adminVdc *govcd.AdminVdc) error {
 
 	_ = d.Set("allocation_model", adminVdc.AdminVdc.AllocationModel)
-	_ = d.Set("cpu_guaranteed", *adminVdc.AdminVdc.ResourceGuaranteedCpu)
-	_ = d.Set("cpu_speed", adminVdc.AdminVdc.VCpuInMhz)
+	if adminVdc.AdminVdc.ResourceGuaranteedCpu != nil {
+		_ = d.Set("cpu_guaranteed", *adminVdc.AdminVdc.ResourceGuaranteedCpu)
+	}
+	if adminVdc.AdminVdc.VCpuInMhz != nil {
+		_ = d.Set("cpu_speed", int(*adminVdc.AdminVdc.VCpuInMhz))
+	}
 	_ = d.Set("description", adminVdc.AdminVdc.Description)
-	_ = d.Set("enable_fast_provisioning", adminVdc.AdminVdc.UsesFastProvisioning)
-	_ = d.Set("enable_thin_provisioning", adminVdc.AdminVdc.IsThinProvision)
+	if adminVdc.AdminVdc.UsesFastProvisioning != nil {
+		_ = d.Set("enable_fast_provisioning", *adminVdc.AdminVdc.UsesFastProvisioning)
+	}
+	if adminVdc.AdminVdc.IsThinProvision != nil {
+		_ = d.Set("enable_thin_provisioning", *adminVdc.AdminVdc.IsThinProvision)
+	}
 	_ = d.Set("enable_vm_discovery", adminVdc.AdminVdc.VmDiscoveryEnabled)
 	_ = d.Set("enabled", adminVdc.AdminVdc.IsEnabled)
-	_ = d.Set("memory_guaranteed", *adminVdc.AdminVdc.ResourceGuaranteedMemory)
+	if adminVdc.AdminVdc.ResourceGuaranteedMemory != nil {
+		_ = d.Set("memory_guaranteed", *adminVdc.AdminVdc.ResourceGuaranteedMemory)
+	}
 	_ = d.Set("name", adminVdc.AdminVdc.Name)
 
-	networkPool, err := govcd.GetNetworkPoolByHREF(vcdClient.VCDClient, adminVdc.AdminVdc.NetworkPoolReference.HREF)
-	if err != nil {
-		return fmt.Errorf("error retrieving network pool: %s", err)
+	if adminVdc.AdminVdc.NetworkPoolReference != nil {
```

Comment:
Just a check - are we safe to assume that if reference is returned - then it is a fact we are admins and this will work.

My thought - if the reference was returned (even if we have no access to it) the call below would fail and "break" read.

Reply:
For org admin in we don't have NetworkPoolReference in XML.

Reply:
Alright, agreed. I was afraid of different vCD versions misbehaving, but our testing process should capture any errors.